modin/core/io/text/text_file_dispatcher.py

```diff
@@ -1057,8 +1092,21 @@ def _read(cls, filepath_or_buffer, **kwargs):
                 newline=newline,
                 header_size=header_size,
                 pre_reading=pre_reading,
+                read_callback_kw=read_callback_kw
+                if not compute_metadata_before_skipping_rows
+                else None,
```

Comment:
you should set `read_callback_kw` to `None` in `if compute_metadata_before_skipping_rows` branch, it would be read easier IMHO